sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/CheckpointPosition.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   The position in a partition's event stream to use when updating a checkpoint, indicating that an event processor should begin reading from the next event.
+    /// </summary>
+    ///
+    public struct CheckpointPosition : IEquatable<CheckpointPosition>
+    {
+        /// <summary>
+        ///   The offset to associate with the checkpoint. This value will be used for informational metadata in almost all cases. If there is no sequence number associated
+        ///   with the checkpoint, this value will be used for positioning when events are read.
+        /// </summary>
+        ///
+        public long Offset { get; }
+
+        /// <summary>
+        ///   The sequence number to associate with the checkpoint. It indicates that a processor should begin reading from the next event in the stream.
```

Comment:
```suggestion
        ///   The sequence number to associate with the checkpoint.
```

Reply:
TODO: I don't currently have the sequence number > offset change in this PR. Think more about whether we should add it now. If not, fix this documentation to match the documentation I left in the old `UpdateCheckpointAsync` overload.